Services/WindowServer/WindowManager.h

```diff
@@ -169,8 +169,21 @@ class WindowManager : public Core::Object {
     void start_window_resize(Window&, const Gfx::IntPoint&, MouseButton);
     void start_window_resize(Window&, const MouseEvent&);
 
-    const Window* active_fullscreen_window() const { return (m_active_window && m_active_window->is_fullscreen()) ? m_active_window : nullptr; }
-    Window* active_fullscreen_window() { return (m_active_window && m_active_window->is_fullscreen()) ? m_active_window : nullptr; }
+    const Window* active_fullscreen_window() const
+    {
+        if (m_active_window && m_active_window->is_fullscreen()) {
+            return m_active_window;
+        }
+        return {};
```

Comment:
doesn't this function return a regular pointer? why are we using `{}` instead of `nullptr`?

Reply:
Yes. I will change that.

The reason for the initial change was that the conversion operator from `std::nullptr_t` to `WeakPtr<T>` is removed so when the `?:` operator is used and returns different types (not convertible) the compiler complains. Note, the `m_active_window` is a `WeakPtr<Window>` and `nullptr` is a `std::nullptr_t`.

Reply:
We can also just leave that file alone, there is no need to rewrite these functions without the ternary operator.

Reply:
It does not work using the ternary operator as I described above. This changeset removes the conversion operator from a `std::nullptr_t` to `WeakPtr<Window>`. Therefore, the ternary operator, which used to return a `std::nullptr_t` getting automatically converted to a `WeakPtr<Window>` no longer converts. A ternary operator must return the same type from each sub-expression, so that is why this needs to not be a ternary operator.

This is exactly one of the issues having the conversion operators causes. It leads to this confusion because the conversion from `std::nullptr_t` to `WeakPtr<Window>` is hidden.

The resulting code using the ternary operator (without optimization) would be to call the `WeakPtr<Window>(std::nullptr_t)` constructor and then call the `WeakPtr<Window>` conversion to a `Window*`, but this is hidden. The result now is that `return nullptr;` simply converts to a `Window*`. No construction of the `WeakPtr<Window>` at all.

Reply:
Ah, that's very subtle. Thanks a lot for explaining it.

Reply:
I agree it is subtle. IMO, this kind of automatic conversion is ugly and is why I want to remove these constructors.